src/daemon/group.js

```diff
@@ -335,10 +335,16 @@ class Group extends EventEmitter {
     // Make sure to send only one response
     const send = once(() => {
       let target = item.target + (item.target.endsWith('/') ? '' : '/') + path
```

Comment:
hmm, it seems like ideally we would just not even parse the path out from the query -- whatever it is we'd just append the whole thing.  but maybe that's too much work to change.

Reply:
It's shorter codewise to do that, but I suspect a bit less clear. When we get here, we know the original target (which unfortunately excludes the query params), and the new host, so it seems clearest just to append the three together I think. Otherwise we need to take the entire raw url, overwrite the host (for the redirect case) and parse out (potentially) the service name (for both cases). But, because the hotel port is configurable, and there might be some additional mapping outside of hotel that would modify the url in an unpredictable way, it seems clearer and safer just to grab the parts we're highly confident are correct and append them into a url that should be correct.

Reply:
Right, I'm saying we should just make it so that when we get here we know the original target, including query params.